app/scripts/lib/url.js

```diff
@@ -89,6 +89,16 @@ function (_) {
       return origin;
     },
 
+    /**
+     * Returns true if given "uri" has HTTP or HTTPS scheme so it is navigable
+     *
+     * @param {String} uri
+     * @returns {boolean}
+     */
+    isNavigable: function (uri) {
+      return /^(http|https):\/\//.test(uri);
```

Comment:
IMHO we don't need too much validation here. if the relier sets a bad redirect_uri it will be their fault.
I don't mind changing this though.